flowmachine/flowmachine/features/subscriber/topup_balance.py

```diff
@@ -215,6 +225,17 @@ def _make_query(self):
             """
             return sql
 
+        if self.statistic in {"mode"}:
+            sql = f"""
+            SELECT DISTINCT ON (subscriber) subscriber, balance AS value
```

Comment:
Can't one just do `mode() WITHIN GROUP(ORDER BY balance) as value`?

Reply:
The `TopUpBalance` class weights the statistics it computes by the time that the subscriber held a given balance. All the other stats in this class are weighted in that way, so I think the mode should also be weighted to ensure consistency. In this case, the mode balance is the balance that was held for most of the time. That is, the balance that shows the highest weight. The postgres built-in `mode` function will only compute the most common balance without regards to the weight.

We can contrast the above with the mode for top up recharge, which will follow exactly what you suggested here.